MediaBrowser.Controller/Entities/BaseItem.cs

```diff
@@ -1607,6 +1607,11 @@ private bool IsVisibleViaTags(User user)
                 return false;
             }
 
+            if (user.GetPreference(PreferenceKind.AllowedTags).Any() && !user.GetPreference(PreferenceKind.AllowedTags).Any(i => Tags.Contains(i, StringComparison.OrdinalIgnoreCase)))
```

Comment:
Should get the preference once.

```suggestion
            var allowedTagsPreference = user.GetPreference(PreferenceKind.AllowedTags);
            if (allowedTagsPreference.Any() && !allowedTagsPreference.Any(i => Tags.Contains(i, StringComparison.OrdinalIgnoreCase)))
            {
                return false;
            }
```